datafusion/functions/Cargo.toml

```
@@ -55,10 +55,11 @@ chrono = { workspace = true }
 datafusion-common = { workspace = true }
 datafusion-execution = { workspace = true }
 datafusion-expr = { workspace = true }
+datafusion-physical-expr = { workspace = true, default-features = true }
 hex = { version = "0.4", optional = true }
 itertools = { workspace = true }
 log = { workspace = true }
-
+regex = { version = "1.8" }
```

Since this is only needed for the regular expressions, I think we can mark it optional like

```suggestion
regex = { version = "1.8", optional = true }
```

And then change this above from

```toml
regex_expressions = []
```

to 

```toml
regex_expressions = ["regex"]
```

I'll do this in a follow on PR

---

https://github.com/apache/arrow-datafusion/pull/9473